Frameworks/CoreText/CTFramesetter.mm

```diff
@@ -30,47 +30,18 @@ - (void)dealloc {
 }
 @end
 
-static _CTFrame* _createFrame(_CTFramesetter* framesetter, CGRect frameSize, CGSize* sizeOut, bool createFrame) {
-    _CTFrame* ret = nil;
-
-    if (createFrame) {
-        ret = [_CTFrame new];
-        ret->_framesetter = framesetter;
-        ret->_frameRect = frameSize;
+static _CTFrame* _createFrame(_CTFramesetter* framesetter, CGRect frameSize, CFRange range) {
+    if (!framesetter) {
+        return nil;
     }
 
     // Call _DWriteWrapper to get _CTLine object list that makes up this frame
     _CTTypesetter* typesetter = static_cast<_CTTypesetter*>(framesetter->_typesetter);
-    CFRange range = CFRangeMake(0, typesetter->_characters.size());
-    NSArray<_CTLine*>* lines = _DWriteGetLines(typesetter, range, frameSize);
-    if (createFrame) {
-        ret->_lines = lines;
-        for (_CTLine* line in static_cast<id<NSFastEnumeration>>(ret->_lines)) {
-            ret->_lineOrigins.emplace_back(line->_lineOrigin);
-        }
+    if (range.length == 0) {
+        range.length = typesetter->_characters.size();
     }
-
-    if (sizeOut && range.length != 0) {
-        // TODO::
-        // Is there a better way to do this - Investigate.
-        // For now the below logic is used to calculate the frame size required to fit the text in -
-        //     width = frame width as it is the constraint passed to DWrite to calculate renderable area width
-        //     height = min(_lineOrigin.y of the last line in the frame + the _lineOrigin.y of the first line (so we atleast get the
-        //              empty space that the first line got), frameSize.size.height)
-
-        // Calculate width
-        sizeOut->width = frameSize.size.width;
-
-        // Calculate height
-        _CTLine* firstLine = static_cast<_CTLine*>([lines objectAtIndex:0]);
-        _CTLine* lastLine = static_cast<_CTLine*>([lines objectAtIndex:[lines count] - 1]);
-        // TODO::
-        // Modify this code once we move away from Cairo Freetype for rasterizing text.
-        CGFloat originYFirstLine = frameSize.size.height - firstLine->_lineOrigin.y - frameSize.origin.y;
-        CGFloat originYLastLine = frameSize.size.height - lastLine->_lineOrigin.y - frameSize.origin.y;
-        sizeOut->height = min(originYLastLine + originYFirstLine, frameSize.size.height);
-    }
-
+    _CTFrame* ret = _DWriteGetFrame(typesetter, range, frameSize);
+    ret->_framesetter = framesetter;
     return ret;
```

Comment:
>ret [](start = 11, length = 3)

[[ret retain] autorelease] - this will guarantee the autorelease object stays until the function that is calling this returns